bundlewrap/repo.py

```diff
@@ -537,6 +543,28 @@ def _build_node_metadata(self, blame=False):
                             )
                         self._node_metadata_partial[node.name] = new_metadata
 
+            # At this point, static metadata from groups and nodes has been merged.
+            # Next, we look at defaults from metadata.py.
+
+            for node_name in self._node_metadata_partial:
```

Comment:
I think this is missing a mechanism like `self._node_metadata_static_complete.add(node_name)` to prevent `@metadata_defaults` from being run again and again.

Reply:
✅